cmd/thanos/query.go

```diff
@@ -257,6 +257,7 @@ func registerQuery(app *extkingpin.App) {
 		Default("0s"))
 
 	blockQueryMetricsWithoutFilter := cmd.Flag("query.block-query-metrics-without-filter", "Comma-separated list of metric patterns to block queries without sufficient label filters. Helps prevent high-cardinality metric queries.").Default("").String()
+	blockQueryBroadRegexPatterns := cmd.Flag("query.block-query-broad-regex-patterns", "Comma-separated list of overly broad regex patterns (like .+, .*, etc.) to block unconditionally in __name__ regex queries. Helps prevent expensive broad regex queries.").Default("").String()
```

Comment:
i wonder if we could combine the flag into a single one, I recall old behavior is `metric_name{}` with empty filter, in the new case, it is similar `.+{}` without filters too, just here `.+` or `.*` is the metric name we try to block?